query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -568,7 +568,17 @@ private void handleAggregationAndGroupBy(
   }
 
   private void handleTableFormatResultSet(
-      ResultSetGroup resultSetGroup, List<Builder> rowBuilderList) {
+      ResultSetGroup resultSetGroup, List<Builder> rowBuilderList,LinkedHashSet<Expression> allSelections) {
+
+    Map<Integer,String >selectionMap= new HashMap<Integer,String>();
+    int selectionCounter=0;
```

Comment:
Can we move to a func that returns a set of avg_rate column indices?
e.g `Set<Integer> getAvgRateColumnIndices(<args>)?